### PR TITLE
Chetan added the redirection part to the current user

### PR DESCRIPTION
**Pull request by Chetan0606**
# Description
Users are unable to navigate back to the main dashboard after viewing another user's profile. 
image

## Related PRS (if any):
This frontend PR is related to the dev backend.
…

## Main changes explained:
- Added the navigation logic to Header.jsx

## How to test:
1. check into the current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. Clear site data/cache
4. log as owner/admin user
5. Go to dashboard→Leaderboard Section→ Click on red/green dot before the username→Click Welcome, {Username}→Click view profile→Click on X on the yellow bar→Click ok
6. verify if you are navigated to your dashboard
7. verify this new feature works in [dark mode](https://docs.google.com/document/d/11OXJfBBedK6vV-XvqWR8A9lOH0BsfnaHx01h1NZZXfI)

## Screenshots or videos of changes:
https://github.com/user-attachments/assets/13ffda49-f610-49f8-b4c2-268203b3a123

### CODE REVIEW

File: src/components/Header/Header.jsx [hunk 1]
```diff
@@ -202,6 +202,7 @@ export function Header(props) {
     sessionStorage.removeItem('viewingUser');
     window.dispatchEvent(new Event('storage'));
     props.getWeeklySummaries(user.userid)
+    window.location.href = `/dashboard/${userId}`;
```
Comment (kabraambika) [inline src/components/Header/Header.jsx:205]: **Suggestion for code improvement**: Using window.location.href reloads the entire application, which can be inefficient. The useHistory hook from React Router is better suited for this case as it enables navigation without a full reload. 

logged-in user ID should not visible in the URL and Url should be '/dashboard' which maintains consistency throughout the application.

**Review by kabraambika — CHANGES_REQUESTED** [review]
I have reviewed this frontend PR and suggested changes.

**Review by kabraambika — APPROVED** [review]
I have reviewed the changes and it's working fine now. Good work!

File: src/components/Header/Header.jsx (hunk 1 above)
Comment (kabraambika) [reply src/components/Header/Header.jsx]: Resolving the conversation as this is fixed in changeset 201ad4f1abf550ff2669fe79c1cb7466e7afe291

**Review by manikittu810 — APPROVED** [review]
Hello @Chetan0606 
Followed the mentioned steps and everything is workin as mentioned.  Below is the screen recording.

https://github.com/user-attachments/assets/27022924-7a10-43a1-baf8-852d723712c2


Thank you.

**Review by SrikanthChowdary495 — APPROVED** [review]
Everything working as expected. Great Work! 

https://github.com/user-attachments/assets/f825d3b6-fd4e-4f83-8481-7e2ef2e7073e

**Review by Pranay-ai — APPROVED** [review]
Hey @Chetan0606 
The PR is Working as Expected 

https://github.com/user-attachments/assets/585a8200-8111-4dcd-a458-d21796bc8eea

Thank You.

**Review by vijay10pk — APPROVED** [review]
This fix is working as expected.
https://www.loom.com/share/ece3ae5146ad42b8a85d5d394483c012?sid=3b24eabd-c8c3-46c8-9575-3a72d71707f7

**Review by DiyaWadhwani — APPROVED** [review]
Hi Chetan,
Went through your changes. All looks good and works as expected.
Thank you for your time with this. I have dropped a working video of the testing below.

https://github.com/user-attachments/assets/ccf42ea5-625a-4ff2-9e28-1f394b1cfaeb

**Review by cgomezhub — COMMENTED** [review]
Hey Chetan!

I see you've implemented useHistory to redirect the button to /dashboard. That's good. However, I don't see a difference between a personal dashboard and a simple dashboard, except for the yellow window.

Screenshot 2024-08-08 070734


Screenshot 2024-08-08 070804

**Review by Logeshwari-Renu — APPROVED** [review]
I have tested the PR and verified the redirection to current user from selected user. The functionality work fine. Great work!

**Review by vishavk1992 — APPROVED** [review]
i have tested this PR and it works as expected, good work


https://github.com/user-attachments/assets/11f14034-b9e6-4f69-b4e0-3bffc2f1bf33

**Review by wang9hu — APPROVED** [review]
Work as intended!

https://github.com/user-attachments/assets/1d08477c-00c5-4e9c-9c24-a235375db05a

**Review by aaryaneil — APPROVED** [review]
Tested the PR and the changes work as expected and the navigation works as described. 

https://github.com/user-attachments/assets/975838dc-f103-46c6-9e1e-cbbcb55465ad

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**